src/main/scala/io/applicative/datastore/util/reflection/ReflectionHelper.scala

```diff
@@ -0,0 +1,95 @@
+package io.applicative.datastore.util.reflection
+
+import java.time.{LocalDateTime, OffsetDateTime, ZonedDateTime}
+import java.util.Date
+
+import com.google.cloud.datastore.{Blob, DateTime, Entity, LatLng}
+import io.applicative.datastore.Key
+import io.applicative.datastore.exception.UnsupportedFieldTypeException
+import io.applicative.datastore.util.DateTimeHelper
+
+import scala.reflect.runtime.universe._
+
+private[datastore] trait ReflectionHelper extends DateTimeHelper {
+
+  private val ByteClassName = classOf[Byte].getSimpleName
+  private val IntClassName = classOf[Int].getSimpleName
+  private val LongClassName = classOf[Long].getSimpleName
+  private val FloatClassName = classOf[Float].getSimpleName
+  private val DoubleClassName = classOf[Double].getSimpleName
+  private val StringClassName = classOf[String].getSimpleName
+  private val JavaUtilDateClassName = classOf[Date].getSimpleName
+  private val BooleanClassName = classOf[Boolean].getSimpleName
+  private val DatastoreDateTimeClassName = classOf[DateTime].getSimpleName
+  private val DatastoreLatLongClassName = classOf[LatLng].getSimpleName
+  private val DatastoreBlobClassName = classOf[Blob].getSimpleName
+  private val LocalDateTimeClassName = classOf[LocalDateTime].getSimpleName
+  private val ZonedDateTimeClassName = classOf[ZonedDateTime].getSimpleName
+  private val OffsetDateTimeClassName = classOf[OffsetDateTime].getSimpleName
+
+  private[datastore] def extractRuntimeClass[E: TypeTag](): RuntimeClass = {
+    val mirror = runtimeMirror(getClass.getClassLoader)
+    mirror.runtimeClass(typeOf[E].typeSymbol.asClass)
+  }
+
+  private[datastore] def instanceToDatastoreEntity[E](key: Key[E], classInstance: E, clazz: Class[_]): Entity = {
+    var builder = Entity.newBuilder(key.key)
+    clazz.getDeclaredFields
+      .filterNot(_.isSynthetic)
+      .tail
```

Comment:
I assume that the first field is ID, which is already encapsulated in Key, so there is no need to add id to the builder.